migrations/lib/importer/steps/badge_groupings.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+module Migrations::Importer::Steps
+  class BadgeGroupings < ::Migrations::Importer::CopyStep
+    MAX_NAME_LENGTH = 100
+    MAX_DESCRIPTION_LENGTH = 500
+
+    store_mapped_ids true
+
+    requires_mapping :ids_by_name, "SELECT LOWER(name), id FROM badge_groupings"
+
+    column_names %i[id name description position created_at updated_at]
+
+    total_rows_query <<~SQL, MappingType::BADGE_GROUPINGS
+      SELECT COUNT(*)
+      FROM badge_groupings
+           LEFT JOIN mapped.ids mapped_badge_grouping
+             ON badge_groupings.original_id = mapped_badge_grouping.original_id
+               AND mapped_badge_grouping.type = ?
+      WHERE mapped_badge_grouping.original_id IS NULL
+    SQL
+
+    rows_query <<~SQL, MappingType::BADGE_GROUPINGS
+      SELECT badge_groupings.*,
+             ROW_NUMBER() OVER (
+              ORDER BY COALESCE(badge_groupings.position, 0), badge_groupings.original_id
+            ) AS normalized_position
+      FROM badge_groupings
+           LEFT JOIN mapped.ids mapped_badge_grouping
+             ON badge_groupings.original_id = mapped_badge_grouping.original_id
+               AND mapped_badge_grouping.type = ?
+      WHERE mapped_badge_grouping.original_id IS NULL
+      ORDER BY normalized_position
+    SQL
+
+    def execute
+      # TODO:(selase) Use @discourse_db
+      @max_position = DB.query_single("SELECT COALESCE(MAX(position), 0) FROM badge_groupings")[0]
```

Comment:
I'd say it's perfectly fine to use `BadgeGrouping.maximum(:position) || 0` in this case. It's a single value, so nothing performance critical.

Reply:
Sure. I was contemplating that but I figured it doesn't hurt  to skip the ActiveRecord overheard anyways